File: client-python/ui/tela_login.py
import time
import sessao
from ui.banner import Banner
from models import LoginRequest, ContaLogada


def executar_login(api_client) -> bool:
    Banner.limpar()
    Banner.exibir_cabecalho("ACESSO À CONTA")
    Banner.titulo_secao("IDENTIFICAÇÃO DO CLIENTE")

    Banner.prompt("CPF          :")
    cpf = input().strip()

    Banner.prompt("Senha        :")
    senha = input().strip()

    Banner.prompt("Tipo da Conta [1-Corrente, 2-Poupança]:")
    try:
        tipo = int(input().strip())
    except ValueError:
        Banner.erro("Tipo de conta inválido. Digite 1 ou 2.")
        _pausa(2000)
        return False

    Banner.espaco()
    Banner.linha_separadora()
    Banner.info("Autenticando...")
    _pausa(500)

    try:
        req = LoginRequest(cpf=cpf, senha=senha, tipoConta=tipo)

        resposta_login = api_client.login(req)

        numero_conta = resposta_login.get("numero") if isinstance(resposta_login, dict) else resposta_login.numero
        response_conta = api_client.obterDados(numero_conta)

        msg_sucesso = resposta_login.get("mensagem") if isinstance(resposta_login, dict) else resposta_login.mensagem
        Banner.sucesso(msg_sucesso)
        _pausa(1000)

        nome_titular = resposta_login.get("nome") if isinstance(resposta_login, dict) else resposta_login.nome

        conta_logada = ContaLogada(
            numero=numero_conta,
            nome=nome_titular,
            cpf=cpf,
            tipo=tipo,
            saldo=response_conta.saldo,
            limite=response_conta.limite,
            rendimento=response_conta.taxa,
        )


        sessao.iniciar(conta_logada)
        return True

    except Exception as e:
        Banner.erro(f"Falha no login: {e}")
        _pausa(2000)
        return False


def _pausa(milissegundos: int):
    time.sleep(milissegundos / 1000.0)

File: client-java/src/main/java/com/example/client_java/ui/TelaLogin.java
package com.example.client_java.ui;

import com.example.client_java.model.ContaLogada;
import com.example.client_java.model.request.LoginRequest;
import com.example.client_java.model.response.ContaResponse;
import com.example.client_java.model.response.LoginResponse;
import com.example.client_java.service.BancoApiClient;
import org.springframework.stereotype.Component;
import org.springframework.web.client.HttpClientErrorException;

import java.util.Scanner;

@Component
public class TelaLogin {
    private final Scanner sc;
    private final BancoApiClient bancoApiClient;

    public TelaLogin(BancoApiClient bancoApiClient, Scanner sc) {
        this.bancoApiClient = bancoApiClient;
        this.sc = sc;
    }

    public ContaLogada exibir() {
        Banner.limpar();
        Banner.exibirCabecalho("ACESSO À CONTA");
        Banner.tituloSecao("IDENTIFICAÇÃO DO CLIENTE");

        System.out.print("CPF          : ");
        String cpf = sc.nextLine().trim();

        System.out.print("Senha        : ");
        String senha = sc.nextLine().trim();

        System.out.print("Tipo da Conta [1-Corrente, 2-Poupança]: ");
        int tipo = Integer.parseInt(sc.nextLine().trim());

        Banner.espaco();
        Banner.linhaSeparadora();
        Banner.info("Autenticando...");
        pausa(500);

        try {
            LoginRequest req = new LoginRequest(cpf, senha, tipo);

            LoginResponse resposta = bancoApiClient.login(req);

            ContaResponse response = bancoApiClient.obterDados(resposta.numero());

            Banner.sucesso(resposta.mensagem());
            pausa(1000);

            return new ContaLogada(
                    response.numero(),
                    cpf,
                    resposta.nome(),
                    tipo,
                    response.saldo(),
                    response.limite(),
                    response.taxa()
            );

        } catch (HttpClientErrorException e) {
            String erro = e.getResponseBodyAsString();
            String mensagem = erro
                    .replaceAll(".*\"erro\":\"*", "")
                    .replaceAll("\"}.*", "")
                    .trim();

            Banner.erro("Falha no login: " + mensagem);
            pausa(2000);

            return null;
        }
    }

    private void pausa(long ms) {
        try { Thread.sleep(ms); } catch (InterruptedException ignored) {}
    }
}
